spring-xd-related-work.tex

```diff
@@ -49,15 +49,65 @@ \subsection{Spring XD and Storm}
 \item Building upon the functional stream processing model, users have the option
 to choose from Reactor\cite{reactor}, Spark Streaming or RxJava APIs to build
 complex data centric applications.
+\item As an alternative to linear streaming data pipelines, Spring XD provides 
+support for directed ayclic graphs (DAG)\cite{dag} through named channels to 
+combine multiple flows upstream or downstream from the message channel. The 
+behavior of the channel can be either queue-based on topic-based. In comparison, 
+Storm also accepts DAG of operations that represent streaming computations.\\*
+
+Here is an example that shows how you can use a named channel to share a data 
+pipeline driven by different input sources.\\*
+
+\verb;queue:foo > file;\\*
+\verb;http > queue:foo;\\*
+\verb;time > queue:foo;\\*
+
+Now if you post data to the http source, you will see that data intermingled 
+with the time value in the file.\\*
+
+\verb;tap:stream:mystream > file;\\*
+\verb;tap:stream:mystream > log;
+
 \end{itemize*}
 
 Storm and Spring XD support many common data sources and middleware;
 for example, the reading and writing of data payloads from Apache Kafka
-is supported in both. A Bolt in Storm is analogous to a
-source in Spring XD, and spouts in Storm are similar to processors and sinks
+is supported in both. A Spout in Storm is analogous to a
+source in Spring XD, and Bolts in Storm are similar to processors and sinks
 in Spring XD. As stream processing frameworks, Storm and Spring XD can be used for
 similar use-cases.
 
+\subsection{Samza}
+Apache Samza\cite{samza} is a distributed stream processing framework that uses 
+Kafka for messaging and YARN for fault-tolerance and other non-functional
+requirements.
+
+The following features differentiate Spring XD and Samza.
+
+\begin{itemize*}
+\item Spring XD provides out of the box adapters to read and write data fragments,
```

Comment:
Not sure about 'data fragments'. We could simply say

"Spring XD provides OOTB sources and sinks for a wide variety of enterprise systems (do we reference them somewhere). In the case of Samza, the main integration point is reading and writing to Kafka topics, which requires the end user to provide a solution for integrating with other types of systems.